receiver/carbonreceiver/protocol/plaintext_parser_test.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protocol
+
+import (
+	"testing"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_plaintextParser_Parse(t *testing.T) {
+	p := PlaintextParser{}
+	tests := []struct {
+		line    string
+		want    *metricspb.Metric
+		wantErr error
+	}{
+		{
+			line: "tst.int 1 1582230020",
+			want: buildMetric(
+				metricspb.MetricDescriptor_GAUGE_INT64,
+				"tst.int",
+				nil,
+				nil,
+				&metricspb.Point{
+					Timestamp: &timestamp.Timestamp{Seconds: 1582230020},
+					Value:     &metricspb.Point_Int64Value{Int64Value: 1},
+				},
+			),
+		},
+		{
+			line: "tst.dbl 3.14 1582230020",
+			want: buildMetric(
+				metricspb.MetricDescriptor_GAUGE_DOUBLE,
+				"tst.dbl",
+				nil,
+				nil,
+				&metricspb.Point{
+					Timestamp: &timestamp.Timestamp{Seconds: 1582230020},
+					Value:     &metricspb.Point_DoubleValue{DoubleValue: 3.14},
+				},
+			),
+		},
+		{
+			line: "tst.int.3tags;k0=v_0;k1=v_1;k2=v_2 128 1582230020",
+			want: buildMetric(
+				metricspb.MetricDescriptor_GAUGE_INT64,
+				"tst.int.3tags",
+				[]string{"k0", "k1", "k2"},
+				[]string{"v_0", "v_1", "v_2"},
+				&metricspb.Point{
+					Timestamp: &timestamp.Timestamp{Seconds: 1582230020},
+					Value:     &metricspb.Point_Int64Value{Int64Value: 128},
+				},
+			),
+		},
+		{
+			line: "tst.int.1tag;k0=v_0 1.23 1582230020",
+			want: buildMetric(
+				metricspb.MetricDescriptor_GAUGE_DOUBLE,
+				"tst.int.1tag",
+				[]string{"k0"},
+				[]string{"v_0"},
+				&metricspb.Point{
+					Timestamp: &timestamp.Timestamp{Seconds: 1582230020},
+					Value:     &metricspb.Point_DoubleValue{DoubleValue: 1.23},
+				},
+			),
+		},
+	}
```

Comment:
Please also add some error cases to cover the parsing errors.

Reply:
Done